worker/app/operations/upload.py

```diff
@@ -30,9 +40,52 @@ def execute(self):
 
         try:
             with SFTPClient(hostname, port, username, private_key) as client:
-                for file in self.working_dir.iterdir():
+                for file in self.files:
                     if file.is_dir():
                         continue
                     client.upload_file(self.remote_working_dir, file)
+
+                    if self.delete:
+                        file.unlink()
         except SSHException as e:
             raise UploadError(code='paramiko.Error', message=str(e))
+
+    @staticmethod
+    def upload(remote_working_dir: str, files: list = None, directory: Path = None, delete: bool = False):
+        """upload either a list of files (Path) or all files of a directory (Path)"""
+
+        if files is None and directory is None:
+            raise ValueError("upload requires either `files` or `directory`")
+
+        # build list of files if invoked with directory
+        if files is None:
+            files = [file for file in directory.iterdir() if not file.is_dir()]
+
+        stats = [{'name': file.name, 'size': file.stat().st_size} for file in files]
+        files_desc = ','.join(['{name} - {size}'.format(**stat) for stat in stats])
```

Comment:
nit: would you consider consolidate these 2 line into one line with python3.7 style string interpolation?

Reply:
not sure what you mean. I switched to literal string for `files_desc` but don't want to combined them as `stats` is useful and returned while `files_desc` is just for logging.